keras/ops/nn_test.py

```diff
@@ -1870,6 +1870,40 @@ def test_ctc_loss(self):
         result = knn.ctc_loss(labels, outputs, label_length, output_length)
         self.assertAllClose(result, np.array([3.4411672, 1.91680186]))
 
+    @pytest.mark.skipif(
+        backend.backend() not in ["tensorflow", "jax"],
+        reason="CTC decode only supported for TF and JAX backends",
+    )
+    def test_ctc_decode(self):
+        inputs = np.array(
+            [
+                [
+                    [0.1, 0.4, 0.2, 0.0],
+                    [0.3, 0.3, 0.4, 0.0],
+                    [0.3, 0.2, 0.4, 0.0],
+                ],
+                [
+                    [0.1, 0.4, 0.7, 0.0],
+                    [0.3, 0.3, 0.4, 0.0],
+                    [0.2, 0.1, 0.1, 0.0],
+                ],
+                [
+                    [0.1, 0.4, 0.2, 0.0],
+                    [0.3, 0.3, 0.2, 0.0],
+                    [0.3, 0.2, 0.4, 0.0],
+                ],
+            ]
+        )
+        labels = np.array([[1, 2], [2, 0], [1, 0]])
+        score_labels = np.array([[-1.2], [-1.3], [-0.4]])
+
+        (decoded,), scores = knn.ctc_decode(
+            inputs, sequence_lengths=[3, 3, 1], strategy="greedy"
```

Comment:
We should also check the different strategies (when available) and the behavior of the different arguments for the beam search strategy.

Reply:
Just added a unit test for beam search when the backend is TF, thanks!